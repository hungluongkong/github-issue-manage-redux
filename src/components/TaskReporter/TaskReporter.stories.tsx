import React from 'react';
import styled from 'styled-components';
import { storiesOf } from '@storybook/react';

import TaskReporter from '.';

import { TaskReportType } from '../../types';
import { PreviewArea, Code, DetailTable } from '../commonStyled';

// Detail of props
const propsDetail: JSX.Element = DetailTable({
  data: [
    { name: 'type', type: 'TaskReportType', default: 'TaskReportType.Success' },
    { name: 'content', type: 'string' },
  ],
});

const PopupWrapper = styled.div`
  * {
    position: static;
  }
`;

storiesOf('Task Reporter', module)
  .add('Task Reporter Types', () => (
    <>
      <h3>Task Reporter Types</h3>
      <PreviewArea >
        <PopupWrapper>
          <TaskReporter type={TaskReportType.Success} content="Success! Hide After 2 seconds" timeoutMS={2000} date={new Date()} />
          <TaskReporter type={TaskReportType.Warning} content="Warning! Hide after 1 second" timeoutMS={1000} date={new Date()} />
        </PopupWrapper>
      </PreviewArea>
      <Code>
        {`
          <TaskReporter type={TaskReportType.Success} content="Success! Hide After 2 seconds" timeoutMS={2000} date={new Date()} />
          <TaskReporter type={TaskReportType.Warning} content="Warning! Hide after 1 second" timeoutMS={1000} date={new Date()} />
        `}
      </Code>
      {propsDetail}
    </>
  ));
