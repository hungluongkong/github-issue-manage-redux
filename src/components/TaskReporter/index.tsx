import React, { useState, useEffect } from 'react';
import styled from 'styled-components';

import { isEqualAllProps } from '../../helpers/propsHelper';

import * as color from '../../theme/color';
import * as metrics from '../../theme/metrics';
import { TaskReportType, TaskReportProps } from '../../types';

const Wrapper = styled.div<{ type: TaskReportType }>`
  position: absolute;
  bottom: 0;
  right: 0;
  margin: 10px;
  display: flex;
  flex-direction: column;
  border: 1px solid ${color.POPUP_BORDER_COLOR};
  width: 500px;
  outline: 0;
  background: ${(props) =>
    props.type === TaskReportType.Success
      ? color.POPUP_BG_COLOR_SUCCESS
      : color.POPUP_BG_COLOR_WARNING};
`;

const PopupBody = styled.p`
  overflow: hidden;
  text-overflow: ellipsis;
  padding: 5px 10px;
`;

const TaskReporter = ({
  content,
  date,
  type = TaskReportType.Success,
  timeoutMS = metrics.POPUP_TIMEOUT,
}: TaskReportProps) => {
  const [isVisible, setIsVisible] = useState(false);

  useEffect(() => {
    setIsVisible(true);

    // Inform popup auto hide
    setTimeout(() => {
      setIsVisible(false);
    }, timeoutMS);
  }, [timeoutMS, date]);

  return (
    <Wrapper
      type={type}
      style={{ visibility: isVisible && content ? 'visible' : 'hidden' }}
    >
      <PopupBody>{content}</PopupBody>
    </Wrapper>
  );
};

export default React.memo(TaskReporter, isEqualAllProps);
