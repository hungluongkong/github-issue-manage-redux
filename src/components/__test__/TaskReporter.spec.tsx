import React from 'react';
import { create } from 'react-test-renderer';
import TaskReporter from '../TaskReporter';

describe('Unit test for IssueList component', () => {
  it('snapshot', () => {
    const issueList = create(<TaskReporter content="task" date={new Date()} />);
    expect(issueList).toMatchSnapshot();
  });
});
