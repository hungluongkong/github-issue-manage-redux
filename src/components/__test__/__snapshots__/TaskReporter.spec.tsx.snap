// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Unit test for IssueList component snapshot 1`] = `
<div
  className="sc-AxjAm iptBje"
  style={
    Object {
      "visibility": "hidden",
    }
  }
  type={0}
>
  <p
    className="sc-AxirZ dlbijh"
  >
    task
  </p>
</div>
`;
